tests/test_enterprise/test_signals.py

```diff
@@ -600,6 +603,117 @@ def test_delete_enterprise_learner_role_assignment_success(self):
         )
         self.assertFalse(learner_role_assignment.exists())
 
+    def test_delete_single_enterprise_learner_role_assignment(self):
+        """
+        Test that when `EnterpriseCustomerUser` record is deleted, `delete_enterprise_learner_role_assignment`
+        also deletes the enterprise learner role assignment associated with the user's previously linked enterprise
+        customer. However. it should *not* delete the enterprise learner role assignment associated with other
+        enterprise customers.
+        """
+        # Create a new EnterpriseCustomerUser record.
+        EnterpriseCustomerUserFactory(
+            user_id=self.learner_user.id,
+            enterprise_customer=self.enterprise_customer,
+        )
+
+        # Create a EnterpriseCustomerUser record such that the user is linked to multiple enterprises.
+        EnterpriseCustomerUserFactory(
+            user_id=self.learner_user.id,
+            enterprise_customer=self.second_enterprise_customer,
+        )
+
+        # Verify that a new learner role assignment is created for both linked enterprise customers.
+        learner_role_assignment = SystemWideEnterpriseUserRoleAssignment.objects.filter(
+            user=self.learner_user,
+            role=self.enterprise_learner_role,
+            enterprise_customer=self.enterprise_customer,
+        )
+        self.assertTrue(learner_role_assignment.exists())
+        second_learner_role_assignment = SystemWideEnterpriseUserRoleAssignment.objects.filter(
+            user=self.learner_user,
+            role=self.enterprise_learner_role,
+            enterprise_customer=self.second_enterprise_customer,
+        )
+        self.assertTrue(second_learner_role_assignment.exists())
+
+        # Delete EnterpriseCustomerUser record.
+        enterprise_customer_user = EnterpriseCustomerUser.objects.get(
+            user_id=self.learner_user.id,
+            enterprise_customer=self.enterprise_customer,
+        )
+        enterprise_customer_user.delete()
+
+        # Verify appropriate learner role assignment is deleted, but the other
+        # learner role assignment remains.
+        learner_role_assignment = SystemWideEnterpriseUserRoleAssignment.objects.filter(
+            user=self.learner_user,
+            role=self.enterprise_learner_role,
+            enterprise_customer=self.enterprise_customer,
+        )
+        self.assertFalse(learner_role_assignment.exists())
+        second_learner_role_assignment = SystemWideEnterpriseUserRoleAssignment.objects.filter(
+            user=self.learner_user,
+            role=self.enterprise_learner_role,
+            enterprise_customer=self.second_enterprise_customer,
+        )
+        self.assertTrue(second_learner_role_assignment.exists())
+
+    def test_unlink_single_enterprise_learner_role_assignment(self):
```

Comment:
nit: I'm not sure this test is necessary? Why would the order of which an organization was removed matter from the perspective of the database/data model?

If I was to suggest a change here I may write a single test that would add 4-5 orgs, test deleting a few of them and verifying the result?